src/Workspaces/Core/Portable/Options/Option`1.cs

```diff
@@ -2,19 +2,25 @@
 
 using System;
 using System.Collections.Immutable;
+using System.Diagnostics;
 
 namespace Microsoft.CodeAnalysis.Options
 {
     /// <summary>
     /// An global option. An instance of this class can be used to access an option value from an OptionSet.
     /// </summary>
-    public class Option<T> : IOption
+    public class Option<T> : IOptionWithOptionGroup
```

Comment:
i would expect all consumers to be using this class and not implementing hteir own.  so i think we could merge IOptionWithOptionGroup into IOption.

Reply:
or we get rid of that interface, introduce a new Option base type for `Option<T>` and `PerLanguage<T>` to derive from (that contains things like 'Group') and have code that cares check for that base type, not hte interface.

Reply:
I thought about all these possible approaches, but decided to take the safer route of no public API changes in this PR. We certainly need some public API changes here, will file a separate bug for it.

---
In reply to: [219247617](https://github.com/dotnet/roslyn/pull/30043#discussion_r219247617) [](ancestors = 219247617)

Reply:
That works for me.  I agree it woudl be nice to not gate this on the long and complex public api discussion.

Reply:
[https://github.com/dotnet/roslyn/issues/30085](https://github.com/dotnet/roslyn/issues/30085)

---
In reply to: [219316024](https://github.com/dotnet/roslyn/pull/30043#discussion_r219316024) [](ancestors = 219316024)